routes/nft.go

```diff
@@ -1338,7 +1340,389 @@ func (fes *APIServer) _bidEntryToResponse(bidEntry *lib.NFTBidEntry, postEntryRe
+	Transaction       *lib.MsgBitCloutTxn
+	TransactionHex    string
+}
+
+func (fes *APIServer) TransferNFT(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := TransferNFTRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("TransferNFT: Error parsing request body: %v", err))
+		return
+	}
+
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("TransferNFT: Error getting utxoView: %v", err))
+		return
+	}
+
+	// Do a simple validation of the requestData.
+	if requestData.NFTPostHashHex == "" {
+		_AddBadRequestError(ww, fmt.Sprintf("TransferNFT: Must include NFTPostHashHex"))
+		return
+
+	} else if requestData.SenderPublicKeyBase58Check == "" {
+		_AddBadRequestError(ww, fmt.Sprintf("TransferNFT: Must include UpdaterPublicKeyBase58Check"))
+		return
+
+	} else if requestData.ReceiverPublicKeyBase58Check == "" {
+		_AddBadRequestError(ww, fmt.Sprintf("TransferNFT: Must include UpdaterPublicKeyBase58Check"))
+		return
+
+	} else if requestData.SerialNumber <= 0 || requestData.SerialNumber > int(utxoView.GlobalParamsEntry.MaxCopiesPerNFT) {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"TransferNFT: SerialNumbers must be between %d and %d, received: %d",
+			1, utxoView.GlobalParamsEntry.MaxCopiesPerNFT, requestData.SerialNumber))
+		return
+
+	}
+
+	// Get the PostHash for the NFT.
+	nftPostHashBytes, err := hex.DecodeString(requestData.NFTPostHashHex)
+	if err != nil || len(nftPostHashBytes) != lib.HashSizeBytes {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"TransferNFT: Error parsing post hash %v: %v",
+			requestData.NFTPostHashHex, err))
+		return
+	}
+	nftPostHash := &lib.BlockHash{}
+	copy(nftPostHash[:], nftPostHashBytes)
+
+	// Get the sender's public key.
+	senderPublicKeyBytes, _, err := lib.Base58CheckDecode(requestData.SenderPublicKeyBase58Check)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("TransferNFT: Problem decoding sender public key: %v", err))
+		return
+	}
+
+	// Get the receiver's public key.
+	receiverPublicKeyBytes, _, err := lib.Base58CheckDecode(requestData.ReceiverPublicKeyBase58Check)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("TransferNFT: Problem decoding receiver public key: %v", err))
+		return
+	}
+
+	// Get the NFT in question so we can do a more hardcore validation of the request data.
+	nftKey := lib.MakeNFTKey(nftPostHash, uint64(requestData.SerialNumber))
+	nftEntry := utxoView.GetNFTEntryForNFTKey(&nftKey)
+	if nftEntry == nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"TransferNFT: Error could not find the NFT an NFT with postHash %v and serialNumber %d",
+			requestData.NFTPostHashHex, requestData.SerialNumber))
+		return
+
+	} else if nftEntry.IsForSale {
+		_AddBadRequestError(ww, fmt.Sprintf("TransferNFT: Cannot transfer NFT that is for sale."))
+		return
+	}
+
```

Comment:
Should we check that the sender owns the NFT somewhere in this function?

Reply:
Added and tested :+1: